### DISCUSSION TITLE
What is best practise for localising not found page?

### DISCUSSION THREAD

**Discussion by lukasbicus** [vercel/next.js#60020] [Help]
### Summary

I'm working on a next js app. I want to create several not-found pages for multiple locales.

I want the path `/en/unknown-path` to lead to the not-found page in English and `/de/unknown-path` to lead to the not-found page in German.

However, according to the [documentation](https://nextjs.org/docs/app/api-reference/file-conventions/not-found#props), the not-found page doesn't accept props. So I'm unable to use `generateStaticParams`

The not-found page is static, so I can't use `useRouter` with locale detection.

Please share your ideas on the topic.

Note: I work with the latest version of nextJs, I'm using the app router (not pages)

**Comment by ananuness** [comment]
You can use [asPath from useRouter](https://nextjs.org/docs/pages/api-reference/functions/use-router) to have the option to change the ui based on the attempted path in the browser:

```js
export default function NotFound() { 
  const { asPath: path } = useRouter();
  
  if (path.includes("/en/")) { 
    return <div>Not found for /en</div>; 
  } 

  return <div>Not found for /de</div>; 
}
```

I hope it helps, if you have any problems, don't hesitate to ask! 😊
